juju_spell/commands/add_user.py

```
@@ -23,6 +26,12 @@ async def execute(self, controller: Controller, *args: Any, **kwargs: Any) -> Di
             display_name=kwargs["display_name"],
         )
 
+        if kwargs.get("acl"):
+            grant_cmd = GrantCommand()
+            grant_result: Result = await grant_cmd.run(controller=controller, **kwargs)
+            if not grant_result.success:
+                return grant_result
```

I think at this point we should raise an error and not return grand result.

---

Ack

---

How this will help to user? You will return only `output = False` why?

---

Not, the output will be `Result(success=False, error=${error-here})`.
Because the `run` function will return Result object, so it is simple logic to return the Result object from grant cmd.

---

You change it, but before the grant command was not catching any error.